python/pylibcudf/pylibcudf/scalar.pyx

```diff
@@ -79,3 +91,31 @@ cdef class Scalar:
         s.c_obj.swap(libcudf_scalar)
         s._data_type = DataType.from_libcudf(s.get().type())
         return s
+
+    @classmethod
+    def from_py(cls, py_val):
+        """Convert a Python standard library object to a Scalar.
+        """
+        cdef DataType dtype
+        if isinstance(py_val, py_bool):
+            dtype = DataType(type_id.BOOL8)
+            c_val = make_numeric_scalar(dtype.c_obj)
+            (<numeric_scalar[cbool]*>c_val.get()).set_value(py_val)
+        elif isinstance(py_val, int):
+            dtype = DataType(type_id.INT64)
+            c_val = make_numeric_scalar(dtype.c_obj)
+            (<numeric_scalar[int64_t]*>c_val.get()).set_value(py_val)
+        elif isinstance(py_val, float):
+            dtype = DataType(type_id.FLOAT64)
+            c_val = make_numeric_scalar(dtype.c_obj)
+            (<numeric_scalar[double]*>c_val.get()).set_value(py_val)
+        elif isinstance(py_val, str):
+            dtype = DataType(type_id.STRING)
+            c_val = make_string_scalar(py_val.encode())
+        else:
+            raise NotImplementedError(f"{type(py_val).__name__} is not supported.")
```

Comment:
I'm not sure if this is a nitpick (or incorrect) but _should_ we distinguish between a type (that can be converted to a scalar) but is not implemented yet and a type that cannot be converted to a scalar? If its the latter, then we should raise TypeError, right?

Reply:
It is a nitpick, but not incorrect. You're right, that would be optimal. The difficulty is in separating the two. We could have this fallback case be a TypeError and then add another NotImplementedError case that checks isinstance on a bunch of types that we expect to support. It'll be very difficult to capture all of them exhaustively, though. We'll probably have some things falling through to TypeError that we actually want to support for a while if we do this.

Reply:
Sure I made the default raise a `TypeError` and created a separate dispatch for types that should be implemented in the future (`NotImplementedError`)